components/org.wso2.carbon.identity.auth.abstractotp/src/main/java/org/wso2/carbon/identity/auth/abstractotp/util/AuthenticatorUtils.java

```diff
@@ -0,0 +1,53 @@
+package org.wso2.carbon.identity.auth.abstractotp.util;
+
+import org.owasp.encoder.Encode;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.auth.abstractotp.internal.AuthenticatorDataHolder;
+import org.wso2.carbon.identity.handler.event.account.lock.exception.AccountLockServiceException;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.auth.abstractotp.constant.AuthenticatorConstants.ErrorMessages.ERROR_CODE_GETTING_ACCOUNT_STATE;
+import static org.wso2.carbon.identity.auth.abstractotp.constant.AuthenticatorConstants.MULTI_OPTION_URI_PARAM;
+
+/**
+ * Utility functions for the authenticator.
+ */
+public class AuthenticatorUtils {
+
+    /**
+     * Check whether a given user account is locked.
+     *
+     * @param user Authenticated user.
+     * @return True if user account is locked.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    public static boolean isAccountLocked(AuthenticatedUser user) throws AuthenticationFailedException {
+
+        try {
+            return AuthenticatorDataHolder.getAccountLockService().isAccountLocked(user.getUserName(),
```

Comment:
Why do we need a method to wrap this single line?

Reply:
Removed the isAccountLocked method. This is being directly called from the the AbstractOTPAuthenticator now.